js/bitforex.js

```diff
@@ -272,37 +272,64 @@ module.exports = class bitforex extends Exchange {
         return this.parseBalance (response);
     }
 
-    async fetchTicker (symbol, params = {}) {
-        await this.loadMarkets ();
-        const market = this.markets[symbol];
-        const request = {
-            'symbol': market['id'],
-        };
-        const response = await this.publicGetApiV1MarketTicker (this.extend (request, params));
-        const data = response['data'];
-        const timestamp = this.safeInteger (data, 'date');
-        return {
+    parseTicker (ticker, market = undefined) {
+        //
+        // {
+        //     "buy":7.04E-7,
+        //     "date":1643371198598,
+        //     "high":7.48E-7,
+        //     "last":7.28E-7,
+        //     "low":7.10E-7,
+        //     "sell":7.54E-7,
+        //     "vol":9877287.2874
+        // }
+        //
+        const symbol = this.safeSymbol (undefined, market);
+        const timestamp = this.safeInteger (ticker, 'date');
+        return this.safeTicker ({
             'symbol': symbol,
             'timestamp': timestamp,
             'datetime': this.iso8601 (timestamp),
-            'high': this.safeNumber (data, 'high'),
-            'low': this.safeNumber (data, 'low'),
-            'bid': this.safeNumber (data, 'buy'),
+            'high': this.safeString (ticker, 'high'),
+            'low': this.safeString (ticker, 'low'),
+            'bid': this.safeString (ticker, 'buy'),
             'bidVolume': undefined,
-            'ask': this.safeNumber (data, 'sell'),
+            'ask': this.safeString (ticker, 'sell'),
             'askVolume': undefined,
             'vwap': undefined,
             'open': undefined,
-            'close': this.safeNumber (data, 'last'),
-            'last': this.safeNumber (data, 'last'),
+            'close': this.safeString (ticker, 'last'),
+            'last': this.safeString (ticker, 'last'),
             'previousClose': undefined,
             'change': undefined,
             'percentage': undefined,
             'average': undefined,
-            'baseVolume': this.safeNumber (data, 'vol'),
+            'baseVolume': this.safeString (ticker, 'vol'),
             'quoteVolume': undefined,
-            'info': response,
+            'info': ticker,
+        }, market, false);
+    }
+
+    async fetchTicker (symbol, params = {}) {
+        await this.loadMarkets ();
+        const market = this.markets[symbol];
+        const request = {
+            'symbol': market['id'],
         };
+        const response = await this.publicGetApiV1MarketTicker (this.extend (request, params));
+        const ticker = this.safeValue (response, 'data');
+        //
+        // {
```

Comment:
The actual response from the exchange should contain the `data` subdictionary that is not shown in the comment below – please include a complete exchange response here.